test/index.coffee

```diff
@@ -0,0 +1,84 @@
+Promise = require 'bluebird'
+{ expect } = require 'chai'
+_ = require 'lodash'
+Docker = require 'dockerode'
+{ DockerProgress } = require 'docker-progress'
+
+gc = require('../lib/index')
+
+SKIP_GC_TEST = process.env.SKIP_GC_TEST? || false
+DOCKER_PATH = process.env.DOCKER_PATH || '/var/run/docker.sock'
+docker = new Docker({ socketPath: DOCKER_PATH, Promise })
+dockerProgress = new DockerProgress({ socketPath: DOCKER_PATH })
+
+promiseToBool = (p) ->
+	p.then(-> return true).catchReturn(false)
+
+describe 'Garbage collection', () ->
+	# This test case is a little weird, it requires that no other images are present on
+	# the system to ensure that the correct one is being removed. Because of this, you
+	# can use the SKIP_GC_TEST env var to inform the test suite not to run this test
+	it 'should remove the LRU image', () ->
+		this.timeout(600000)
+		return Promise.resolve() if SKIP_GC_TEST
+		# first pull some images, so we know in which order they are referenced
+
+		# Create the promises with a standard map so that they're done in the correct
+		# order (bluebird doesn't guarantee this)
+		pullPromises = [
+			'alpine:3.1'
+			'debian:squeeze'
+			'ubuntu:lucid'
+		].map (image) ->
+			dockerProgress.pull(image, _.noop)
+
+		Promise.all pullPromises
+		.then ->
+			# Attempt to remove a single byte, which will remove the LRU image,
+			# which should be alpine
+			gc.garbageCollect(1)
+		.then ->
+			# Check if the alpine image has been removed
+			docker.getImage('alpine:3.1')
+			.inspect()
+			.then ->
+				Promise.reject('Alpine image still present!')
+			.catch (e) ->
+				# Also check that the other images are still present
+				Promise.all [
+					docker.getImage('debian:squeeze').inspect()
+					docker.getImage('ubuntu:lucid').inspect()
+				]
+
+	it 'should remove more than one image if necessary', () ->
+		this.timeout(600000)
+		return Promise.resolve() if SKIP_GC_TEST
+
+		pullPromises = [
+			'alpine:3.1'
+			'debian:squeeze'
+			'ubuntu:lucid'
+		].map (image) ->
+			dockerProgress.pull(image, _.noop)
+
+		# Get the size of the first image, so we can add one to it to remove
+		# the next one in addition
+		Promise.all(pullPromises)
+		.then ->
+			docker.getImage('alpine:3.1')
+			.inspect()
+		.then (img) ->
+			img.Size + 1
+		.then (size) ->
+			gc.garbageCollect(size)
+		.then ->
+			Promise.all [
+				docker.getImage('alpine:3.1').inspect()
+				docker.getImage('debian:squeeze').inspect()
+				docker.getImage('ubuntu:lucid').inspect()
+			].map(promiseToBool)
+		.then (imgs) ->
+			success = _.isEqual(imgs, [false, false, true])
+			return if success then Promise.resolve() else Promise.reject('Incorrect amount of images deleted')
```

Comment:
```coffee
if not _.isEqual(imgs, [false, false, true])
	throw new Error('Incorrect amount of images deleted')
```
- this avoids having to create/return promises inline, and also makes sure we reject with an Error rather than a string